src/lambda/commands/copyLambdaUrl.ts

```diff
@@ -0,0 +1,26 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DefaultLambdaClient, LambdaClient } from '../../shared/clients/lambdaClient'
+import { LambdaFunctionNode } from '../explorer/lambdaFunctionNode'
+import globals from '../../shared/extensionGlobals'
+import { copyToClipboard } from '../../shared/utilities/messages'
+import { addCodiconToString } from '../../shared/utilities/textUtilities'
+
+export const NO_URL_INFO_MSG = 'No URL for Lambda function.'
+
+export async function copyLambdaUrl(
+    node: LambdaFunctionNode,
+    client: LambdaClient = new DefaultLambdaClient(node.regionCode),
+    copyToClipboardFunc: (data: string, label?: string) => Promise<void> = copyToClipboard
+): Promise<void> {
+    const url = await client.getFunctionUrl(node.name)
+    if (!url) {
```

Comment:
Alternatively this could be part of the lambda node info, and the `Copy URL` action could be grayed-out (`enablement=false`) if there is no URL for the given lambda.

Reply:
This was one of the discussions I was having with Josh and this is what we concluded.

### To implement Justins approach
- [Update this function](https://github.com/nkomonen-amazon/aws-toolkit-vscode/blob/b346e3b1e3520b9f0bb291127faebf6700649835/src/lambda/explorer/lambdaNodes.ts#L74) and the method above `updateInPlace()`. This will be to allow us to pass in an object that has the Function URL info (`FunctionUrlConfig`).
- Update how [this property works](https://github.com/nkomonen-amazon/aws-toolkit-vscode/blob/b346e3b1e3520b9f0bb291127faebf6700649835/src/lambda/explorer/lambdaNodes.ts#L74) by creating an `updateContextValue()` method which would take a list of values and then join them. This will allow us to not affect the existing features but we can add additional values such as a string that looks like `"awsRegionFunctionNode/withFunctionUrl"`
- Then in the [`ViewItem`](https://github.com/nkomonen-amazon/aws-toolkit-vscode/blob/b346e3b1e3520b9f0bb291127faebf6700649835/package.json#L1487) pattern we'd need to update all the existing patterns from `/^(awsRegionFunctionNode|awsRegionFunctionNodeDownloadable)$/"` to `/(awsRegionFunctionNode|awsRegionFunctionNodeDownloadable)/"` for the existing ones.

There is some extra work to implement this. I can more commits to this PR or open up a new one following this.

---

For Jaden's I agree to adding in a link to set up the Lambda URL if it doesn't exist.

Reply:
> Update how [this property works](https://github.com/nkomonen-amazon/aws-toolkit-vscode/blob/b346e3b1e3520b9f0bb291127faebf6700649835/src/lambda/explorer/lambdaNodes.ts#L74) by creating an updateContextValue()

Hopefully we can avoid `updateContextValue()` and just change `updateChildren()` to get the lambda info and set a `hasUrl` property.

But yes, let's do that in a followup PR.